astropy/io/ascii/core.py

```diff
@@ -1466,7 +1472,13 @@ def _get_writer(Writer, fast_writer, **kwargs):
     if 'exclude_names' in kwargs:
         writer.exclude_names = kwargs['exclude_names']
     if 'fill_values' in kwargs:
-        writer.data.fill_values = kwargs['fill_values']
+        # Prepend user-specified values to the class default.
+        with suppress(TypeError, IndexError):
+            # Test if it looks like (match, replace_string, optional_colname),
+            # in which case make it a list
+            kwargs['fill_values'][1] + ''
+            kwargs['fill_values'] = [kwargs['fill_values']]
+        writer.data.fill_values = kwargs['fill_values'] + writer.data.fill_values
```

Comment:
This is (potentially) a change in behaviour. Previously, user defined fill values would replace writer defaults, now they are added to writer defaults. I believe that that's not a big issue for our built-in classes, because we only have defaults for None, missing, empty etc. However, I have build classes for my own use that derive from BaseReader and replace "-999" with missing. However, my collaborators sometimes also send me tables where missing value are marked with other numbers. In that case I used to call my own class with `fill_value='+999'`. In the new implementation, this would not treat both +999 and -999 as missing. For my particular case all data value are between -10 and +10, so I'm fine either way, I just wanted to give an example where is might actually matter. 

To summarize: I think this change (very few people build their own reader classes) is OK, but should be mentioned in the change log.